prawcore/exceptions.py

```diff
@@ -146,6 +146,30 @@ class TooLarge(ResponseException):
     """Indicate that the request data exceeds the allowed limit."""
 
 
+class TooManyRequests(ResponseException):
+    """Indicate that the user has sent too many requests in a given amount of time."""
+
+    def __init__(self, response):
+        """Initialize a TooManyRequests exception instance.
+
+        :param response: A requests.response instance that may contain a retry-after
+        header and a message.
+
+        """
+        self.response = response
+        self.retry_after = response.headers.get("retry-after")
+        self.message = response.text  # Not all response bodies are valid JSON
+
+        msg = f"received {response.status_code} HTTP response"
+        msg += (
+            f". Please wait at least {float(self.retry_after)} seconds "
+            "before re-trying this request."
+            if self.retry_after
+            else ""
+        )
```

Comment:
```suggestion
        if self.retry_after:
            msg += f". Please wait at least {float(self.retry_after)} seconds before re-trying this request."
```

Reply:
I can do that. I patterned it after the message in Redirect class, which also could be changed.